pkg/kubelet/kubelet.go

```diff
@@ -1009,6 +1010,9 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		// AppArmor is a Linux kernel security module and it does not support other operating systems.
 		klet.appArmorValidator = apparmor.NewValidator()
 		handlers = append(handlers, lifecycle.NewAppArmorAdmitHandler(klet.appArmorValidator))
+	} else if goos == "windows" {
+		// PodLevelResources feature is not supported for Windows
+		handlers = append(handlers, lifecycle.NewPodLevelResourcesHandler())
```

Comment:
I'd like to take a slightly different approach to this:

1. Create a new generic handler for checking feature compatibility, maybe something like `PodFeaturesAdmitHandler`.
2. Create a `features_{linux,windows,unsupported}.go` in the lifecycle directory, similar to what was in https://github.com/kubernetes/kubernetes/pull/132627/files#diff-00fbb8cb4fcf4ea260d7497ed65513ff4ac77278797330b0ec6383e474c9b579 (don't reuse the one in allocation_manager though)
3. The generic admission handler should check if pod-level resources are set, and if so call the features helper to see whether they're supported.

Does that make sense?

Reply:
Thanks! I tried making the changes. Is this along the lines of what you were expecting? If so, I'll add unit tests.